src/modules/ens/sagas.ts

```diff
@@ -0,0 +1,129 @@
+import { Eth } from 'web3x-es/eth'
+import { call, put, takeEvery, select } from 'redux-saga/effects'
+import {
+  GET_ENS_REQUEST, 
+  GetENSRequestAction, 
+  getENSSuccess, 
+  getENSFailure, 
+  SET_ENS_REQUEST,
+  SetENSRequestAction, 
+  setENSSuccess, 
+  setENSFailure
+} from './actions'
+import { ENS } from 'contracts/ENS'
+import { ENSResolver } from 'contracts/ENSResolver'
+import { Address } from 'web3x-es/address'
+import { getAddress } from 'decentraland-dapps/dist/modules/wallet/selectors'
+import { 
+  ENS_ADDRESS, 
+  ENS_RESOLVER_ADDRESS 
+} from 'modules/common/contracts'
+import { ipfs } from 'lib/api/ipfs'
+import { namehash } from "@ethersproject/hash";
+import * as contentHash from 'content-hash'
+import { ENS_EMPTY_CONTENT, ENS_EMPTY_RESOLVER } from './constants'
+
+export function* ensSaga() {
+  yield takeEvery(GET_ENS_REQUEST, handleGetENSRequest)
+  yield takeEvery(SET_ENS_REQUEST, handleSetENSRequest)
+}
+
+function* handleSetENSRequest(action: SetENSRequestAction) {
+  const { land, ens } = action.payload
+  try {
+    const [eth, from] = yield getEth()
+    const nodehash = namehash(ens)
+    const ensContract = new ENS(eth, Address.fromString(ENS_ADDRESS))
+    const resolverContract = new ENSResolver(eth, Address.fromString(ENS_RESOLVER_ADDRESS))
+
+    const ipfsHash = yield call(() => ipfs.uploadRedirectionFile(land))
+    const hash = contentHash.fromIpfs(ipfsHash)
+    yield call(() => 
+      resolverContract.methods
+        .setContenthash(nodehash, `0x${hash}`)
+        .send({from})
+        .getTxHash()
+    )
+    yield call(() => 
+      ensContract.methods
+        .setResolver(nodehash, Address.fromString(ENS_RESOLVER_ADDRESS))
+        .send({from})
+        .getTxHash()
```

Comment:
I think we might want to keep track of these transactions right? show them on the /activity page and/or if they're loading to disable controls or show a notice. To do that we can use `buildTransactionPayload` like we do on `land/actions.ts`

@cazala @nachomazzara am I missing the mark? I confess I'm looking at the code without trying it out yet

Reply:
IMO, we should separate the saga in two different transactions. One for setting the resolver and after confirmation (mined) the second one to set the content. Adding both to the activity page using `buildTransactionPayload`.

Reply:
I agree, the `setResolver` transaction should be handled in a separate saga, and only dispatch that saga before this one if the resolver needs to be set. That way wee can return both `txHash` in each success action and keep track of each transaction individually.

Reply:
Agree with using 2 sagas, in some point I made it without `buildTransactionPayload` on purpose. I didn't find a way to combine two txHash in one saga to use it as payload in the action.  
*Working on it!*